modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -545,19 +548,47 @@ class NDB_Form_Statistics extends NDB_Form
         } else {
             $Param_Project = '';
         }
-        $subprojects = Utility::getSubprojectsForProject($currentProject);
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
 
-        $this->tpl_data['Subprojects'] = $subprojects;
         foreach ($subprojects as $key=>$val) {
             $subprojList .= $key.",";
         }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubProject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if (isset($_REQUEST['BehaviouralSubproject'])
+            && $_REQUEST['BehaviouralSubproject'] != ''
+            && in_array($_REQUEST['BehaviouralSubproject'], array_keys($subprojects))
+        ) {
+            $currentSubProject = $_REQUEST['BehaviouralSubproject'];
+            $this->tpl_data['CurrentSubproject']
+                = array(
+                   'ID'   => $currentSubProject,
+                   'Name' => $subprojects[$currentSubProject],
+                  );
+            $suproject_query
+                = "AND (s.SubprojectID IS NULL".
+                " OR s.SubprojectID = ($currentSubProject))";
```

Comment:
You're right, I missed the in_array, that does likely protect it from injection, but the fact that it's from a dropdown doesn't really matter. Hackers have access to curl.

Regardless of the in_array, it would still be far safer to use a prepared statement. I believe the project original was written before the prepared select wrappers existed in LORIS (and if it didn't, that should have been caught in review too). Any equality check in a `WHERE` clause should use a prepared parameters. The only exception is `IN` clauses, because you can't bind an array to a prepared statement variable (which is really annoying.)